Golem/Jobs.cs

```diff
@@ -67,35 +74,23 @@ public void SetAllJobsFinished()
 
     public async Task<List<IJob>> List(DateTime since)
     {
-        if(_yagna == null || _yagna.HasExited)
+        if (_yagna == null || _yagna.HasExited)
             throw new Exception("Invalid state: yagna is not started");
 
         var agreementInfos = await _yagna.Api.GetAgreements(since);
         var invoices = await _yagna.Api.GetInvoices(since);
 
-        foreach(var agreementInfo in agreementInfos.Where(a => a!=null && a.AgreementID!=null))
+        foreach (var agreementInfo in agreementInfos.Where(a => a != null && a.AgreementID != null))
         {
-            var agreement = await _yagna.Api.GetAgreement(agreementInfo.AgreementID!);
-            var activities = await _yagna.Api.GetActivities(agreement!.AgreementID!);
-            var invoice = invoices.FirstOrDefault(i => i.AgreementId == agreement.AgreementID);
-            foreach(var activityId in activities)
-            {
-                var usage = await _yagna.Api.GetActivityUsage(activityId);
-                if(usage.CurrentUsage != null)
-                {
-                    var price = GetPriceFromAgreementAndUsage(agreement, usage.CurrentUsage);
-                    await UpdateJob(
-                        activityId,
-                        invoice,
-                        price!=null ? new GolemUsage(price) : null);
-                }
-            }            
+            await UpdateJobStatus(agreementInfo.Id);
+            await UpdateJobUsage(agreementInfo.Id);
+
+            var invoice = invoices.FirstOrDefault(i => i.AgreementId == agreementInfo.Id);
+            if (invoice != null)
+                await UpdateJobPayment(invoice);
         }
 
-        return _jobs
-            .Values
-            .Where(j => j.Timestamp>=since)
-            .Cast<IJob>().ToList();
+        return _jobs.Values.Where(job => job.Timestamp >= since).OrderByDescending(job => job.Timestamp).Cast<IJob>().ToList();
```

Comment:
Sorting previously was bit vague, but what I noticed is that previously the last executed job got appended to the end of the list, and now it appears at the top.
Do we want to keep `Descending` ordering?

Reply:
It was consciously changed. I think you should get newest jobs first.
Do you see reasons to sort differently?

Reply:
no